### DISCUSSION TITLE
Failed to load SuitCLib.dll (Ben Bridge 0.83)

### DISCUSSION THREAD

**Discussion by bridgeworld** [lorserker/ben#150] [General]
![2024121106](https://github.com/user-attachments/assets/04cc760b-1fdf-493b-9586-2db4b75612e6)
![2024121505](https://github.com/user-attachments/assets/15b5d26f-77f0-47c1-981b-eb11bbbb1ea6)

OS: Windows 10 LTSC 21H2 Chinese

**Comment by ThorvaldAagaard** [comment]
Most common error for this is Windows blocking DLL's from the internet
Interesting here the first 2 DLL's is loaded.

Please click on properties for the file, and veridy it is not blocked.

Please check, in both locations the file can be (It is included twice as the way it is being loaded is different ondifferent platforms)

So best is to make a search

There is a change in next version, but if nothing work you can disable SuitC in the file in the config directory

**Comment by ThorvaldAagaard** [comment]
I have created a new version. Let me know how it go.

**Comment by bridgeworld** [comment]
Thank you for your reply, it almost worked, only there was an error in the last step. No cards are dealt.
![2024121602](https://github.com/user-attachments/assets/ddb7afe3-53f3-43f5-9ea0-3f76cb74f265)

https://github.com/user-attachments/assets/4c38eb31-dfc4-498b-a448-d629fe95a06d

**Comment by ThorvaldAagaard** [comment]
Ups, I can see there is misisng a file.
Create directry BBA, and in that create directory, and then place this file in that folder
https://github.com/lorserker/ben/blob/main/BBA/CC/BEN-SAYC.bbsa

![image](https://github.com/user-attachments/assets/e59cc67f-b134-4663-b53e-1eeb0fd733ae)

It is missing in the zip-file

I will add it later, but need to check why there was no error during startup

**Reply by bridgeworld** [reply]
I did it according to your method, but the problem still persists.

**Comment by ThorvaldAagaard** [comment]
Ok, I will release 3.4 later today, then you can try. This time the scrolling is working, so we can see the error if any

**Reply by bridgeworld** [reply]
That's great!
Can you provide simple instructions for the use of BBA and MvsM?

**Reply by ThorvaldAagaard** [reply]
BBA is for using BEN together with Bridge Bidding Analyser: https://sites.google.com/view/bbaenglish

Follow instructions on the site to set up BEN as a bidder in that system. No play only bidding

MvsM is preparation for playing a small tournament between different robots, so primarily used for tournament organizer.

If you just would like to play you can join JBridge, where BEN is playing in the tournaments

**Reply by bridgeworld** [reply]
Thank you for your kindness. I often use BBO and FunBridge to play bridge, as well as various software such as Jack/Q-Plus/GIB/WBridge5. I have also used BridgeMonitor to organize several bridge software battles. Of course, i pay more attention to the development progress of Ben.
![2024121712](https://github.com/user-attachments/assets/f38c0be7-cbbd-4f9e-a8c6-62342792cb0e)

In addition, after installing BEN v0.8.0.alpha, it can be used normally, but subsequent versions do not work.
<img width="960" alt="2024121806" src="https://github.com/user-attachments/assets/52d05794-1391-4278-ba39-8fd281eeb5f2" />
<img width="960" alt="2024121803" src="https://github.com/user-attachments/assets/262ed358-7d87-4ec5-86be-0446fd324084" />

**Comment by ThorvaldAagaard** [comment]
All should work if you just clone the repository. Just my experiments with GUI executables, that is taking some time to fix, so new version still not available. I need to test all versions.

By the way you should use appserver.py from the frontend directory

Looking forward to some interesting error reports from you about bidding and play :-)

**Comment by ThorvaldAagaard** [comment]
Please try version 0.8.4

**Reply by bridgeworld** [reply]
I couldn't find the download link.

**Reply by ThorvaldAagaard** [reply]
Sorry, forgot to publsih. Done now

**Reply by bridgeworld** [reply]
Unfortunately, for the same machine, 0.80 is OK, but 0.84 still cannot dealt cards.
<img width="586" alt="2024121908" src="https://github.com/user-attachments/assets/b28eb138-6ece-4ce5-9165-83f7c8f9f5fc" />
https://github.com/user-attachments/assets/22a53499-97e8-4567-89a8-0211913dd459
https://github.com/user-attachments/assets/30473c9c-0298-4df7-a5ef-43a3177dae3e

**Comment by ThorvaldAagaard** [comment]
When the server is started it should look like this
![image](https://github.com/user-attachments/assets/975937b0-f0fb-4fd9-a758-2fb52b6f7778)

Looks like, what i can see in the video

When I click on play i get this
![image](https://github.com/user-attachments/assets/82d456de-1649-4f7e-8c81-26d28b749f26)

Noticed it is a reduced version, where you do not select the server and only enter a board number.

I assume the page in the video is from the previous version, where you can select the server (BEN allows multiple servers with different system)

On your page I can see
![image](https://github.com/user-attachments/assets/163fae9c-1dd3-46e3-91e2-62e770e2c61c)

So it will try to access a server at port 4441 - but the server is listening at port 4443.

So when you get on the first page in the browser click Ctrl-F5 to reload the page, and you should see the reduced page.

You can also play with the full page, but then you will have to select the right server: Default (21GF)

Let me know how it goes

**Comment by bridgeworld** [comment]
Thank you very much. The BEN can now run normally, but some minor bugs have been found. Please refer to the video and screenshots for details.


https://github.com/user-attachments/assets/6421ad82-283b-4296-8e79-6eeb7c9efdb1

<img width="954" alt="2024122001" src="https://github.com/user-attachments/assets/56bf682b-b045-4ab3-a8f0-7aaab035d2d0" />
<img width="954" alt="2024122003" src="https://github.com/user-attachments/assets/b81d418a-ad16-4d45-8751-c747bd14b127" />
![2024122006](https://github.com/user-attachments/assets/1d8e0404-8ae9-4151-961e-88e986b4d26a)
![2024122007](https://github.com/user-attachments/assets/74042657-0891-4745-b05c-09d2c03cd3ac)
![2024122008](https://github.com/user-attachments/assets/bd613098-60b0-458a-98a3-e1d2b114ff8d)

**Reply by ThorvaldAagaard** [reply]
1: I see the screen with many options, but if you are fine with that I should perhaps make it an option. Actually it is the home button, that needs an update.
2: Preventing delete is intentionally, when BEN is published on the internet, as someone liked to delete boards just for fun, but that protection make no sense, when plying locally, so I will enable that - you can always just delete the gamedb-files to start from scratch.

**Reply by ThorvaldAagaard** [reply]
I think I have fixed the errors, but will not release before next version if that is OK?
I can make a version available if needed.

**Reply by bridgeworld** [reply]
Thanks! NEXT version OK.
